vestapol/web_resources/base_resource.py

```diff
@@ -28,7 +33,7 @@ def __init__(
         self.request_headers = request_headers
         self.requested_at = DateTime.utcnow().replace(microsecond=0)
 
-    def load(self, destination):
+    def load(self, destination: DestinationTypes):
```

Comment:
Would the following work?
```suggestion
    def load(self, destination: type[BaseDestination]):
```

Got the idea from [this post](https://adamj.eu/tech/2021/05/16/python-type-hints-return-class-not-instance/) (though it may require some extra logic for compatibility with 3.8)

Reply:
Yeah this was the first thing I tried, I saw that post as well. It did not work unfortunately. mypy continued to create errors like cannot be type `Local` expects `type[BaseDestination]`. May be because of incompatibilities with the superclass

Reply:
Yeah I managed to get it to work on everything except for the functions in `json_writer` and `text writer`, where I got the weird errors below. Let's go with the `DestinationTypes`/`ResourceTypes`

```
❯ poetry run mypy vestapol/ tests/
/Users/waligob/.pyenv/versions/3.9.7/lib/python3.9/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
  warnings.warn(
vestapol/writers/text_writer.py:5: error: Missing positional argument "self" in call to "write_data" of "BaseDestination"
vestapol/writers/json_writer.py:11: error: Missing positional argument "pathname" in call to "write_data" of "BaseDestination"
vestapol/writers/json_writer.py:11: error: Argument 1 to "write_data" of "BaseDestination" has incompatible type "str"; expected "BaseDestination"
vestapol/writers/json_writer.py:16: error: Missing positional argument "pathname" in call to "write_data" of "BaseDestination"
vestapol/writers/json_writer.py:16: error: Argument 1 to "write_data" of "BaseDestination" has incompatible type "str"; expected "BaseDestination"
Found 5 errors in 2 files (checked 20 source files)
```

Reply:
yup i saw the same. Doesn't work when the classes are actually instantiated. Got errors when running the example code in the README as well